client/src/partials/vouchers/complex.js

```diff
@@ -0,0 +1,261 @@
+angular.module('bhima.controllers')
+.controller('ComplexJournalVoucherController', ComplexJournalVoucherController);
+
+ComplexJournalVoucherController.$inject = [
+  'VoucherService', '$translate', 'AccountService',
+  'CurrencyService', 'SessionService', 'FindEntityService',
+  'FindReferenceService', 'NotifyService'
+];
+
+/**
+ * Complex Journal Vouchers
+ *
+ * This module implements complex journal vouchers. It allows users to quickly create transactions by
+ * specifying two or more lines of transactions and all relative document references
+ *
+ * @constructor
+ *
+ * @todo - Implement caching mechanism for incomplete forms (via AppCache)
+ * @todo - Implement Patient Invoices data and Cash Payment data for modal
+ * @todo - Implement a mean to categorise transactions for cashflow reports 
+ */
+function ComplexJournalVoucherController(Vouchers, $translate, Accounts, Currencies, Session, FindEntity, FindReference, Notify) {
+  var vm = this;
+
+  // bread crumb paths
+  vm.paths = [{
+    label : $translate.instant('VOUCHERS.COMPLEX.TITLE'),
+    current: true
+  }];
+
+  // bind the startup method as a reset method
+  vm.submit = submit;
+  vm.currencySymbol     = currencySymbol;
+  vm.addVoucherItem     = addVoucherItem;
+  vm.removeVoucherItem  = removeVoucherItem;
+  vm.checkRowValidity   = checkRowValidity;
+  vm.selectAccount      = selectAccount;
+  vm.openEntityModal    = FindEntity.openModal;
+  vm.openReferenceModal = FindReference.openModal;
+
+  // load the list of accounts
+  Accounts.read()
+  .then(function (accounts) {
+    vm.accounts = accounts.map(function (item) {
+      // remove children accounts
+      delete item.children;
+      return item;
```

Comment:
Are you sure accounts still returns `children`?  I think I removed it in a previous pull request.  If it does, this code is fine.  If not, please change these lines to be:
```js
.then(function (accounts) {
  vm.accounts = accounts;
});
```